agent/php_minit.c

```diff
@@ -38,6 +38,27 @@
 #include "fw_laravel.h"
 #include "lib_guzzle4.h"
 
+#include "php_execute.h"
+
+#if ZEND_MODULE_API_NO >= ZEND_8_0_X_API_NO /* PHP8+ */
+/*
+ * Register the begin and end function handlers with the Observer API.
+ */
+static zend_observer_fcall_handlers nr_php_fcall_register_handlers(
+    zend_execute_data* execute_data) {
+  zend_observer_fcall_handlers handlers = {NULL, NULL};
+  if (NULL == execute_data) {
+    return handlers;
+  }
+  if (NULL == execute_data->func) {
+    return handlers;
+  }
```

Comment:
Maybe more of a question than a comment- what does execute_data do in this function? What cases do we expect to hit the NULL statement?

Reply:
`execute_data` is the information that PHP passes us regarding the executing function.
It's type `zend_execute_data` and is defined here: https://github.com/php/php-src/blob/master/Zend/zend_compile.h#L540

We don't anticipate it ever giving us a NULL pointer for this, but because this data structure is what ALL of our instrumentation is based off of, it's good to doublecheck that it isn't NULL :)

Reply:
Do we only need to include `execute_data` here to determine if we should return `handlers` before assigning the `begin` and `end` functions? 
Or will there be additional functionality surrounding `execute_data` added to this function in the future?

Reply:
It won't likely be used much more in this function, but it is passed to the handlers, and all of our instrumentation use the value extensively.  If you do a search for NR_EXECUTE_PROTO  (where it is used in a function signature) or NR_EXECUTE_ORIG_ARGS (where is is passed as a parameter) you can get an idea of the scope of it.